### ISSUE TITLE
Hungarian language pull request

### ISSUE THREAD

**Issue by kiskunk** [Project-OSRM/osrm-text-instructions#271]
I would like to request the hungarian language to be included in the OSRM-text-instructions.

**Comment by yuryleb** [comment]
Hungarian translation on [Transifex](https://www.transifex.com/project-osrm/osrm-text-instructions/language/hu/) needs to complete 164 strings yet. Maybe you could fill them there and check existing strings for correctness?

**Comment by kiskunk** [comment]
> Hungarian translation on [Transifex](https://www.transifex.com/project-osrm/osrm-text-instructions/language/hu/) needs to complete 164 strings yet. Maybe you could fill them there and check existing strings for correctness?

I will, but how can I check for correctness? I mean if I think some is correct, how can I mark it correct?

**Comment by yuryleb** [comment]
Just don't touch such correct string(s) or press Approve or something like this :wink:

**Comment by kiskunk** [comment]
> Hungarian translation on [Transifex](https://www.transifex.com/project-osrm/osrm-text-instructions/language/hu/) needs to complete 164 strings yet. Maybe you could fill them there and check existing strings for correctness?

I checked the existing ones and I filled the missing ones. I do not have review rights on Transifex though, but I would say the language strings are now pullable. :)

**Comment by yuryleb** [comment]
I tried to merge your translation but found many missing(?) whitespaces after `{way_name}`, `{waypoint_name}`, `{distance}` and other macros like below:
```json
        "continue": {
            "default": {
                "default": "Forduljon {modifier}",
                "name": "Forduljon {modifier} és maradjon a(z) {way_name}szakaszon",
                "destination": "Forduljon {modifier} a(z) {destination} felé",
                "exit": "Forduljon be {modifier} a(z) {way_name} irányába"
            },
```
Could you please check this more on Transifex?

And what `a(z)` here mean? Alternate preposition/ending depending on used macro value? Maybe Hungarian translation also needs for [grammar support](https://github.com/Project-OSRM/osrm-text-instructions/blob/master/languages/grammar/README.md)?

**Comment by kiskunk** [comment]
I will check those tommorrow, thanks for the info. And a(z) means that if a word starts with a vowel then the grammar requires "az", if not, then the grammar requires "a" for examle "az Alley street" or "a Melbourne way", but its not that simple because this breaks apart when it comes to codenames instead of street names for example interstate names because it would be "az M0" instead of "a M0" and I don't think its possible to implement, or if so I don't really know the way from reading the grammar support. Also for hungarian grammar it would be grammatically correct if the street names could get the proper ending for example the way_name would be "XY utca" (XY street), and when the english says: "turn left onto XY street", the hungarian would go "fordulj balra az xy utcába", so "utca" from way_name would be modified into "utcába" and we have many different way_name endings like street/road/park/circle etc. each with its own rule of ending.

**Comment by kiskunk** [comment]
I fixed the whitespaces.

**Comment by yuryleb** [comment]
Thank you, now it's all right with white spaces.

There is the only last issue - I suppose the u-turn modifier should start also with small letter:
```json
            "modifier": {
                "left": "balra",
                "right": "jobbra",
                "sharp left": "élesen balra",
                "sharp right": "élesen jobbra",
                "slight left": "enyhén balra",
                "slight right": "enyhén jobbra",
                "straight": "egyenesen",
                "uturn": "Forduljon vissza"
            },
```

**Comment by kiskunk** [comment]
Oh sorry, fixed :)

**Comment by yuryleb** [comment]
@kiskunk, please review created PR with your translation.

Let's discuss possible Hungarian grammar adding after merge.

**Comment by 1ec5** [comment]
Fixed in #274.
